python/examples/workload/example.py

```diff
@@ -0,0 +1,181 @@
+"""
+Workload example: Element-wise sum of two (M, N) float32 arrays on CPU.
+"""
+import numpy as np
+from mlir import ir
+from mlir.runtime.np_to_memref import get_ranked_memref_descriptor
+from mlir.dialects import func, linalg, bufferization
+from mlir.dialects import transform
+from functools import cached_property
+from lighthouse import Workload
+from lighthouse.utils.mlir import (
+    apply_registered_pass,
+    canonicalize,
+    cse,
+    match,
+)
+from lighthouse.utils.execution import (
+    lower_payload,
+    execute,
+    benchmark,
+)
+
+
+class ElementwiseSum(Workload):
+    """
+    Computes element-wise sum of (M, N) float32 arrays on CPU.
+
+    We can construct the input arrays and compute the reference solution in
+    Python with Numpy.
+
+    We use @cached_property to store the inputs and reference solution in the
+    object so that they are only computed once.
+    """
+
+    def __init__(self, M, N):
+        self.M = M
+        self.N = N
+        self.dtype = np.float32
+        self.context = ir.Context()
+        self.location = ir.Location.unknown(context=self.context)
```

Comment:
Left a comment elsewhere about it being mostly possible to elide thinking about `Context`s and unknown `Location`s in most code.